### ISSUE TITLE
How to know which content to render

### ISSUE THREAD

**Issue by kof** [JonnyBGod/famous-infinitescroll#2]
Right now you emit infiniteScroll on update ... the information actually needed to render correct data is the index of row to be rendered.

**Comment by JonnyBGod** [comment]
Can you describe your use case for me to understand why you would need the current index?

I will accept pull requests if you make a good point for the feature.

**Comment by kof** [comment]
Oh somehow surfaces.length (your demo http://codepen.io/JonnyBGod/pen/mCots) is the index of the item reqiured to be rendered ....

**Comment by JonnyBGod** [comment]
surfaces is just an array of 'new Surface()' used in scrollview.sequenceFrom(surfaces);

You don't really need to worry about rendering the items as the scrollview takes care of this for you.

All you need to do is to add your content to the surfaces array and it will be rendered.

**Comment by kof** [comment]
Yes I know, but you have to know where to start from in you data array .... as you don't know which direction has been scrolled and so which items need to be rendered

**Comment by kof** [comment]
I am talking about mapping from potential infinite data array, which can be fetched when required partially, to the representation.

**Comment by kof** [comment]
Just had a discussion on irc #famous channel. 

We need to add a documentation for offset and margin (I know margin  is famous thing) ... because its not easy to get it.

And the biggest issue is the scroll back process and memory management.

At some point you need to start removing data and surfaces out of the memory.
And when you start doing so, when user scrolls back you need to know what page to fetch again and consider margin and offset.

**Comment by JonnyBGod** [comment]
To accomplish that you will need to write your own implementation of the scrollview.

Removing items from the sequence at this point would result on unexpected behaviour for the user as the rendered items would change for the current page.

Thats out of the scope for this library.

But I believe it is simple to extend the scrollview to remove the items outside the viewport from the dom.
Consider that manipulating the dom has performance costs.

**Comment by kof** [comment]
Wait, I am not talking about dom items. Famous reuses dom items already now. I am talking about surfaces array and data array cleanup.

**Comment by kof** [comment]
If I put 200 Surfaces into sequenced array, there are still rendered items to fill the scrollview + margin.

**Comment by kof** [comment]
Not all 200 surfaces map directly 1:1 to dom element as far I understand.

**Comment by JonnyBGod** [comment]
I'll have a look into it next week. Kind of busy at the moment.
Meanwhile fell free to make your own pull request.
